src/apps/workspaces/domain/workspace.py

```diff
@@ -86,6 +91,28 @@ def decrypt_nickname(cls, values):
 
         return values
 
+    @root_validator
+    def decrypt_firstname(cls, values):
+        firstname = values.get("subject_first_name")
+        if firstname:
+            firstname = StringEncryptedType(Unicode, get_key).process_result_value(
+                firstname, dialect=PGDialect_asyncpg.name
+            )
+            values["subject_first_name"] = str(firstname)
+
+        return values
+
+    @root_validator
+    def decrypt_lastname(cls, values):
+        lastname = values.get("subject_last_name")
+        if lastname:
+            lastname = StringEncryptedType(Unicode, get_key).process_result_value(
+                lastname, dialect=PGDialect_asyncpg.name
+            )
+            values["subject_last_name"] = str(lastname)
+
+        return values
+
```

Comment:
Would it be reasonable to combine this into a single `@root_validator`? (This is Copilot's suggestion btw  😅)

```python
@root_validator
def decrypt_fields(cls, values):
    fields_to_decrypt = ["respondent_nickname", "subject_first_name", "subject_last_name"]
    
    for field in fields_to_decrypt:
        value = values.get(field)
        if value:
            value = StringEncryptedType(Unicode, get_key).process_result_value(
                value, dialect=PGDialect_asyncpg.name
            )
            values[field] = str(value)

    return values
```

Reply:
Yeah, I think that's a good suggestion 👌